junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/MethodTestDescriptor.java

```diff
@@ -108,27 +109,31 @@ public SkipResult shouldBeSkipped(JupiterEngineExecutionContext context) throws
 	@Override
 	public JupiterEngineExecutionContext execute(JupiterEngineExecutionContext context,
 			DynamicTestExecutor dynamicTestExecutor) throws Exception {
+		invokeBetweenBeforeAndAfter(context, c -> invokeTestMethod(c, dynamicTestExecutor));
+		return context;
+	}
+
+	protected void invokeBetweenBeforeAndAfter(JupiterEngineExecutionContext context,
+			Consumer<JupiterEngineExecutionContext> action) {
```

Comment:
The term `action` here is insufficient to convey its purpose.

Something like `testInvoker` or `testExecutor` would be more suitable.